Build failure: pkgsCross.mingwW64.pkg-config-unwrapped

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

-

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
libconfig.h:62:28: note: format string is defined here
   62 | #define G_GSIZE_FORMAT "I64u"
  CC       giowin32.lo
  CC       gspawn-win32.lo
gstdio.c: In function 'g_stat':
gstdio.c:496:31: error: passing argument 2 of '_wstat64i32' from incompatible pointer type [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types8;;]
  496 |   retval = _wstat (wfilename, buf);
      |                               ^~~
      |                               |
      |                               GStatBuf * {aka struct stat *}
In file included from gstdio.c:27:
/nix/store/1pkim64f6fqqnyqsrz8s8mhx4gck6jp0-x86_64-w64-mingw32-gcc-14.3.0/x86_64-w64-mingw32/sys-include/sys/stat.h:82:75: note: expected 'struct _stat64i32 *' but argument is of type 'GStatBuf *' {aka 'struct stat *'}
   82 |   _CRTIMP int __cdecl _wstat64i32(const wchar_t *_Name,struct _stat64i32 *_Stat);
      |                                                        ~~~~~~~~~~~~~~~~~~~^~~~~
make[6]: *** [Makefile:1637: libglib_2_0_la-gstdio.lo] Error 1
make[6]: *** Waiting for unfinished jobs....
gtimezone.c: In function 'rule_from_windows_time_zone_info':
gtimezone.c:565:3: warning: 'strncpy' output may be truncated copying 32 bytes from a string of length 63 [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-stringop-truncation-Wstringop-truncation8;;]
  565 |   strncpy (rule->std_name, (gchar*)tzi->StandardName, NAME_SIZE - 1);
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gtimezone.c:566:3: warning: 'strncpy' output may be truncated copying 32 bytes from a string of length 63 [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-stringop-truncation-Wstringop-truncation8;;]
  566 |   strncpy (rule->dlt_name, (gchar*)tzi->DaylightName, NAME_SIZE - 1);
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gthread-win32.c: In function 'g_private_get_impl':
gthread-win32.c:340:16: warning: cast from pointer to integer of different size [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-pointer-to-int-cast-Wpointer-to-int-cast8;;]
  340 |   DWORD impl = (DWORD) key->p;
      |                ^
gthread-win32.c:345:14: warning: cast from pointer to integer of different size [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-pointer-to-int-cast-Wpointer-to-int-cast8;;]
  345 |       impl = (DWORD) key->p;
      |              ^
gthread-win32.c:370:54: error: passing argument 1 of '_InterlockedCompareExchangePointer' from incompatible pointer type [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types8;;]
  370 |               if (InterlockedCompareExchangePointer (&g_private_destructors, destructor,
      |                                                      ^~~~~~~~~~~~~~~~~~~~~~
      |                                                      |
      |                                                      GPrivateDestructor * volatile* {aka struct _GPrivateDestructor * volatile*}
In file included from /nix/store/1pkim64f6fqqnyqsrz8s8mhx4gck6jp0-x86_64-w64-mingw32-gcc-14.3.0/x86_64-w64-mingw32/sys-include/winnt.h:27,
                 from /nix/store/1pkim64f6fqqnyqsrz8s8mhx4gck6jp0-x86_64-w64-mingw32-gcc-14.3.0/x86_64-w64-mingw32/sys-include/minwindef.h:163,
                 from /nix/store/1pkim64f6fqqnyqsrz8s8mhx4gck6jp0-x86_64-w64-mingw32-gcc-14.3.0/x86_64-w64-mingw32/sys-include/windef.h:9,
                 from /nix/store/1pkim64f6fqqnyqsrz8s8mhx4gck6jp0-x86_64-w64-mingw32-gcc-14.3.0/x86_64-w64-mingw32/sys-include/windows.h:69,
                 from glib-init.h:32,
                 from gthread-win32.c:46:
/nix/store/1pkim64f6fqqnyqsrz8s8mhx4gck6jp0-x86_64-w64-mingw32-gcc-14.3.0/x86_64-w64-mingw32/sys-include/psdk_inc/intrin-impl.h:1759:58: note: expected 'void * volatile*' but argument is of type 'GPrivateDestructor * volatile*' {aka 'struct _GPrivateDestructor * volatile*'}
 1759 | void *_InterlockedCompareExchangePointer(void *volatile *Destination, void *ExChange, void *Comperand) {
      |                                          ~~~~~~~~~~~~~~~~^~~~~~~~~~~
gthread-win32.c:376:59: error: passing argument 2 of '_InterlockedCompareExchangePointer' makes pointer from integer without a cast [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wint-conversion-Wint-conversion8;;]
  376 |           if (InterlockedCompareExchangePointer (&key->p, impl, NULL) != NULL)
      |                                                           ^~~~
      |                                                           |
      |                                                           DWORD {aka long unsigned int}
/nix/store/1pkim64f6fqqnyqsrz8s8mhx4gck6jp0-x86_64-w64-mingw32-gcc-14.3.0/x86_64-w64-mingw32/sys-include/psdk_inc/intrin-impl.h:1759:77: note: expected 'void *' but argument is of type 'DWORD' {aka 'long unsigned int'}
 1759 | void *_InterlockedCompareExchangePointer(void *volatile *Destination, void *ExChange, void *Comperand) {
      |                                                                       ~~~~~~^~~~~~~~
gthread-win32.c: In function 'g_thread_xp_get_condition_variable':
gthread-win32.c:823:46: error: passing argument 1 of '_InterlockedCompareExchangePointer' from incompatible pointer type [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types8;;]
  823 |       if (InterlockedCompareExchangePointer (cond, result, NULL) != NULL)
      |                                              ^~~~
      |                                              |
      |                                              GThreadXpCONDITION_VARIABLE * volatile*
/nix/store/1pkim64f6fqqnyqsrz8s8mhx4gck6jp0-x86_64-w64-mingw32-gcc-14.3.0/x86_64-w64-mingw32/sys-include/psdk_inc/intrin-impl.h:1759:58: note: expected 'void * volatile*' but argument is of type 'GThreadXpCONDITION_VARIABLE * volatile*'
 1759 | void *_InterlockedCompareExchangePointer(void *volatile *Destination, void *ExChange, void *Comperand) {
      |                                          ~~~~~~~~~~~~~~~~^~~~~~~~~~~
make[6]: *** [Makefile:1833: libglib_2_0_la-gthread-win32.lo] Error 1
gutils.c: In function 'g_win32_get_system_data_dirs_for_module':
gutils.c:1945:10: warning: 'hmodule' may be used uninitialized [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wmaybe-uninitialized-Wmaybe-uninitialized8;;]
 1945 |       if (hmodule != NULL)
      |          ^
gutils.c:1861:11: note: 'hmodule' was declared here
 1861 |   HMODULE hmodule;
      |           ^~~~~~~
make[6]: Leaving directory '/build/pkg-config-0.29.2/glib/glib'
make[5]: *** [Makefile:1935: all-recursive] Error 1
make[5]: Leaving directory '/build/pkg-config-0.29.2/glib/glib'
make[4]: *** [Makefile:954: all] Error 2
make[4]: Leaving directory '/build/pkg-config-0.29.2/glib/glib'
make[3]: *** [Makefile:1047: all-recursive] Error 1
make[3]: Leaving directory '/build/pkg-config-0.29.2/glib'
make[2]: *** [Makefile:771: all] Error 2
make[2]: Leaving directory '/build/pkg-config-0.29.2/glib'
make[1]: *** [Makefile:707: all-recursive] Error 1
make[1]: Leaving directory '/build/pkg-config-0.29.2'
make: *** [Makefile:463: all] Error 2
```

### Additional context

_No response_

### System metadata

x86_64-linux

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/pk/pkg-config-unwrapped/package.nix
+++ b/pkgs/by-name/pk/pkg-config-unwrapped/package.nix
@@ -65,7 +65,11 @@ stdenv.mkDerivation rec {
     # Silence "incompatible integer to pointer conversion passing 'gsize'" when building with Clang.
     lib.optionals stdenv.cc.isClang [ "-Wno-int-conversion" ]
     # Silence fprintf format errors when building for Windows.
-    ++ lib.optionals stdenv.hostPlatform.isWindows [ "-Wno-error=format" ]
+    ++ lib.optionals stdenv.hostPlatform.isWindows [
+      "-Wno-incompatible-pointer-types"
+      "-Wno-int-conversion"
+      "-Wno-error=format"
+    ]
   );
 
   enableParallelBuilding = true;

